pkg/query/logical/plan_indexscan.go

```diff
@@ -121,11 +123,12 @@ func (i *indexScan) Execute(ec executor2.ExecutionContext) ([]data.Entity, error
 			chunkSet = chunks
 		} else {
 			// afterwards, it must not be nil
-			chunkSet = chunkSet.HashIntersect(chunks)
+			_ = chunkSet.Intersect(chunks)
 		}
 	}
 
-	return ec.FetchEntity(*i.traceMetadata, chunkSet, series.ScanOptions{
+	//TODO: pass correct shardID
+	return ec.FetchEntity(*i.traceMetadata, 0, chunkSet, series.ScanOptions{
```

Comment:
@lujiajing1126 As mentioned above, once getting the searching results from a particular shard, you would fetch entities from that shard's trace series.